pkg/tsdb/elasticsearch/client/client.go

```diff
@@ -207,7 +207,7 @@ func (c *baseClientImpl) createMultiSearchRequests(searchRequests []*SearchReque
 			header: map[string]interface{}{
 				"search_type":        "query_then_fetch",
 				"ignore_unavailable": true,
-				"index":              c.indices,
```

Comment:
Something we should ideally also do is to dedupe the array. With a "daily" pattern and the index as `[logs-]YYYY.MM.DD,[test],[metrics-]YYYY.MM.DD` we would correctly end up in multiple `[logs-]YYYY.MM.DD` and `[metrics-]YYYY.MM.DD` indices, but `test` will also be duplicated, i.e. ending in a total of `logs-2023.07.05,test,metrics-2023.07.05,logs-2023.07.06,test,metrics-2023.07.06,logs-2023.07.07,test,metrics-2023.07.07` for 3 days.